misc/Kconfig

```diff
@@ -44,9 +44,14 @@ endif
 config TEXT_SECTION_OFFSET
 	hex "TEXT section offset"
 	default 0
+	default 0x200 if BOOTLOADER_MCUBOOT
 	help
-	  This sets and offset before the text section and allows booting images
-	  with custom headers that are inserted at the beginning of the image.
+	  If the application is built for chain-loading by MCUboot using the
+	  BOOTLOADER_MCUBOOT option, a suitable default value for
+	  TEXT_SECTION_OFFSET is set with sufficient space to store the image
+	  header, which should also meet vector table alignment requirements
+	  on most ARM targets. Other targets may be able to use smaller values,
+	  or require larger ones.
```

Comment:
Can you keep this generic? this is applicable to ANY bootloader, not only mcuboot

Reply:
Do you mean the help text or setting the default value to 0x200?

Reply:
@nashif let me know what you think. I tried to be generic and then add a precision for the MCUboot case.